### DISCUSSION TITLE
socketify + aiohttp?

### DISCUSSION THREAD

**Discussion by bitnom** [aio-libs/aiohttp#7994] [Q&A]
I know it's possible to use a custom socket function with aiohttp. Has anyone tried using socketify in this way or know any reason why it wouldn't work?

**Comment by Dreamsorcerer** [comment]
It appears to be a completely different framework, not a socket function? Unless I'm looking at the wrong thing, I don't see how you would combine them (or why you'd want to).

**Reply by bitnom** [reply]
I assume there might be a performance benefit if possible, though this could be ignorance on my part. It's not a socket function but I was thinking maybe it could be rigged to implement as one, or one of its primitives could be. Whether or not this would confer any speedups is obviously questionable as it's more than just a socket (for one). I've not compared multiple socket implementations to know whether or not it makes much of a difference.

**Reply by Dreamsorcerer** [reply]
Again, it appears to be a framework that is competing with aiohttp, not a library designed to integrate with other frameworks. It's like asking to use Django with aiohttp. I just don't see how it makes sense, or how you'd even want such a thing to look like.

If socketify have come up with something that can improve performance without sacrificing features, then it would be good if someone can investigate whether we can take anything from there and improve aiohttp. Regardless, I suspect there's plenty of performance improvements that can be made to aiohttp if someone wants to spend some time profiling the code and finding the bottlenecks. We already have several performance improvements landing for the future 3.10 release (some will be a dramatic difference for a subset of users).

**Reply by Dreamsorcerer** [reply]
Hmm, under their benchmarks for WSGI/ASGI, they do list things like 'fastapi[socketify]`, so maybe there is some kind of integration. The benchmarks make it sound like it could be a replacement for uvicorn, but I'm less clear from the documentation on how that's supposed to work. As we don't use ASGI, I don't know if it would work with aiohttp though. Feel free to take a deeper look, maybe it might be possible to do something similar to the gunicorn integration: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/worker.py
